### PR TITLE
Uscore invariant fix

### PR DESCRIPTION
**Pull request by arscan**
`xyz.where(expression).exists()` would improperly evaluate to true when the `where` expression returned `[]`.  An expression like this is used in us core now, which exposed the error.

This PR also updates versions in the local Gemfile.lock (shouldn't affect users of the gem, unless you run the tests, etc).

### CODE REVIEW

**Comment by arscan** [comment]
I don't expect you to have time @jawalonoski, but FYI.  I suspect that perhaps we want the `where` to shoot out `:null` instead of `[]` when it matches on nothing, but updating `exists` to also return false against `[]` seems safer for someone that isn't familiar with the details of how this is implemented.

We use the fhir_models fhirpath implementation in some of our internal inferno unit tests, and us core was just updated to expose this issue.

**Comment by jawalonoski** [comment]
I think either approach would work. I would choose the one that breaks the least number of other invariants in the unit tests.

File: lib/fhir_models/fhirpath/evaluate.rb [hunk 1]
```diff
@@ -287,7 +287,7 @@ def self.compute(tree, data)
             tree[index] = (previous_node == :null || previous_node.empty? rescue previous_node.nil?)
             clean_index(tree, previous_index)
           when :exists
-            tree[index] = !previous_node.nil? && previous_node != :null
+            tree[index] = !previous_node.nil? && previous_node != :null && previous_node != []
```
Comment (Jammjammjamm) [inline lib/fhir_models/fhirpath/evaluate.rb:290]: Do we need to check for `{}` as well? If not, this seems good to me.

Comment (arscan) [reply lib/fhir_models/fhirpath/evaluate.rb:290]: Good thought, but I have no idea if it is possible to get a `{}` here.  I think it is good enough to just scope this fix to the one we came across for now.

**Pull request merged**